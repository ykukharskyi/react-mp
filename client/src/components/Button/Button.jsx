import React from 'react';

import Input from './Button.styles';

const Button = () => (
  <Input type="button" value="Some title for the button" />
);

export default Button;
