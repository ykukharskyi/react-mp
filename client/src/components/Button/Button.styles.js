import styled from 'styled-components';

export default styled.input`
  background-color: #e50914;
  color: #fff;
  padding: 7px 17px;
  font-weight: 400;
  border: none;
  border-radius: 3px;
  font-size: 16px;
`;
