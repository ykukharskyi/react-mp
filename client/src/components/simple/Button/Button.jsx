import React from 'react';
import PropTypes from 'prop-types';

import Input from './Button.styles';

const Button = ({ title }) => (
  <Input type="button" value={title} />
);

Button.propTypes = {
  title: PropTypes.string.isRequired
};

export default Button;
