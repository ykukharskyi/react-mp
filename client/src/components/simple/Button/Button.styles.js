import styled from 'styled-components';

export const PrimaryButton = styled.input`
  width: 140px;
  height: 35px;
  background-color: #e50914;
  color: #fff;
  font-weight: 400;
  border: none;
  border-radius: 3px;
  font-size: 16px;
  text-transform: uppercase;
`;

export const PrimarySmallButton = styled.input`
  width: 70px;
  height: 22px;
  background-color: #e50914;
  color: #fff;
  font-weight: 600;
  border: none;
  border-radius: 3px;
  font-size: 12px;
  text-transform: uppercase;
`;

export const PrimarySmallGreyButton = styled.input`
  width: 70px;
  height: 22px;
  background-color: #808080;
  color: #fff;
  font-weight: 600;
  border: none;
  border-radius: 3px;
  font-size: 12px;
  text-transform: uppercase;
`;

export const SecondaryButton = styled.input`
  width: 140px;
  height: 35px;
  background-color: #fff;
  color:  #e50914;
  font-weight: 400;
  border: none;
  border-radius: 3px;
  font-size: 16px;
  text-transform: uppercase;
`;

export const TransparentButton = styled.input`
  background-color: Transparent;
  color: #000;
  font-weight: 500;
  border: none;
  font-size: 12px;
`;
