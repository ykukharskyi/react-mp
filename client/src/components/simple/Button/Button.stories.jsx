import React from 'react';
import { storiesOf } from '@storybook/react';
import Button from './Button';

storiesOf('Button', module)
  .add('primary', () => <Button theme="primary" title="Search" />)
  .add('secondary', () => <Button theme="secondary" title="Search" />)
  .add('transparent', () => <Button theme="transparent" title="release date" />)
  .add('primary small', () => <Button theme="primary-small" title="title" />)
  .add('primary small grey', () => <Button theme="primary-small-grey" title="title" />);
