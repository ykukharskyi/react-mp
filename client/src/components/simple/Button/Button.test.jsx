import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';

describe("Button", () => {
  it("primary", () => {
    const wrapper = shallow(<Button theme="primary" title="Search" onClick={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("secondary", () => {
    const wrapper = shallow(<Button theme="secondary" title="Search" onClick={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("transparent", () => {
    const wrapper = shallow(<Button theme="transparent" title="release date" onClick={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("primary small", () => {
    const wrapper = shallow(<Button theme="primary-small" title="title" onClick={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("primary small grey", () => {
    const wrapper = shallow(<Button theme="primary-small-grey" title="title" onClick={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
