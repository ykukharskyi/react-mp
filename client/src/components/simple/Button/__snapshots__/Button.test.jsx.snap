// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button primary 1`] = `
<styled.input
  onClick={[Function]}
  type="button"
  value="Search"
/>
`;

exports[`Button primary small 1`] = `
<styled.input
  onClick={[Function]}
  type="button"
  value="title"
/>
`;

exports[`Button primary small grey 1`] = `
<styled.input
  onClick={[Function]}
  type="button"
  value="title"
/>
`;

exports[`Button secondary 1`] = `
<styled.input
  onClick={[Function]}
  type="button"
  value="Search"
/>
`;

exports[`Button transparent 1`] = `
<styled.input
  onClick={[Function]}
  type="button"
  value="release date"
/>
`;
